CKSIncrementalStore/CKSIncrementalStoreSyncOperation.swift

```diff
@@ -524,7 +524,7 @@ class CKSIncrementalStoreSyncOperation: NSOperation {
                     return true
                 })
                 var values = ckRecord.dictionaryWithValuesForKeys(keys)
-                managedObject.setValuesForKeysWithDictionary(values)
+                managedObject.setValuesForKeysWithDictionary(values) // TODO: it would be safer to filter the keys first to ensure they exist in managedObject. Over time schema changes could result in fields that only on the server - this was covered in one of the videos from WWDC 2014.
```

Comment:
The problem comes when you release version 2 of the app and add a new attribute.  If a user upgrades their iPhone but not their iPad then the app essentially has 2 different active schemas.  When V2 uploads a record with the new attribute it will be synced across to V1 of the app which will try to set the value of the new attribute when it doesn't exist, resulting in a crash.  Removing an attribute has a similar effect.  So unless you are handling that earlier in the  process this will be a problem.